java/src/main/java/ai/rapids/cudf/MemoryListener.java

```diff
@@ -84,22 +84,45 @@
   public abstract void deallocation(long amount, long id);
 
   /**
-   * Holds the set of all listeners. An <code>AtomicReference&lt;HashSet&gt;</code> is used here to
-   * optimize the common read path over the less common write path (add/remove listeners). Other
-   * data structures like a ConcurrentHashMap still involve a lock on the read path where as this
-   * eliminates any locking on reads.
+   * Holds the set of all device listeners. An <code>AtomicReference&lt;HashSet&gt;</code> is used
+   * here to optimize the common read path over the less common write path (add/remove listeners).
+   * Other data structures like a ConcurrentHashMap still involve a lock on the read path where as
+   * this eliminates any locking on reads.
    */
-  private static final AtomicReference<HashSet<MemoryListener>> listeners =
+  private static final AtomicReference<HashSet<MemoryListener>> deviceListeners =
       new AtomicReference<>(new HashSet<>());
 
   /**
-   * Register a memory listener. If the memory listener is already registered it will not install
-   * duplicates. This should be done before any column operations happen or you risk missing those
-   * operations.
+   * Holds the set of pinned memory listeners. An <code>AtomicReference&lt;HashSet&gt;</code> is
```

Comment:
This isn't just for pinned memory -- it's for all HostMemoryBuffer whether they are using pinned memory or not, correct?